### DISCUSSION TITLE
应该如何给第三方组件提issue或pr？

### DISCUSSION THREAD

**Discussion by timongh** [the1812/Bilibili-Evolved#2747] [综合讨论 / General]
**背景**

目前整合第三方组件有两种方式（详见 [CONTRIBUTING][contributing]）
1. 直接将第三方组件代码以 pr 的形式提交到官方仓库
2. 提交 pr 时，不提交代码主体，仅提交自己的插件信息和 GitHub 用户名

**问题**

“视频内显示时间”（内部名称：videoCurTime）组件是由作者 @FoundTheWOUT 创建的，似乎采用了上述第二种整合方式，于是出现了以下两个问题。
1. 无法向其提交 issue。在其仓库中没有“Issues”按钮（可能是作者没有开启该功能）。

    <img width="500" alt="no-issues-btn" src="https://user-images.githubusercontent.com/46739861/146689474-b4bb6fe0-c14d-48db-bec6-9e9f13239a8c.png">

2. 找不到向其提交 pr 的方式。点击“New Pull Request”按钮会被重定向到官方仓库。然而在这里搜索不到该作者的仓库，原因未知。

    <img width="500" alt="no-pr" src="https://user-images.githubusercontent.com/46739861/146689482-d2a2defa-da24-4306-9426-a57590109266.png">

在这种情况下，如果直接向官方仓库提 issue，该作者无法收到通知。如果自行修改，并直接向官方仓库提 pr，覆盖掉原组件作者的提交，似乎又不太好。

因此，是否有一个更好的解决方案？或者说，这其中有什么地方是我理解错误的？如果有，希望能指出。

[contributing]: https://github.com/the1812/Bilibili-Evolved/blob/736e248b1f6b3a68dd6827ef29372c37a4af55d7/CONTRIBUTING.md#自行保留

**Comment by the1812** [comment]
确实挺奇怪, 这里看起来能选择 Fork 的仓库, 但是却没有 @FoundTheWOUT 的那个
![image](https://user-images.githubusercontent.com/26504152/146699262-91630860-bb39-4657-b568-d38b7d8f3fd7.png)

**Reply by FoundTheWOUT** [reply]
我在自己的仓库点开是有的，你试试搜索一下？🤣

**Reply by timongh** [reply]
找到提 pr 的方法了：https://github.com/the1812/Bilibili-Evolved/discussions/2747#discussioncomment-1844734

**Comment by FoundTheWOUT** [comment]
~~之前没有开 issue 的必要，所以我关了~~
issue 我已经打开了，至于 PR，我觉得你可以尝试一下在我的仓库的 PR 页面点 New。(好像在自己的仓库点也可以)
![image](https://user-images.githubusercontent.com/47922588/146699247-a38e206d-9c23-4247-bc14-8ab2760334cf.png)
但我的仓库历史会和本仓库的历史有点不同，所以你可能需要开一个新的，和我仓库历史一样的分支

还有就是，其实你可以选择在这个仓库开 issue 或者 discussion，然后 @ 我一下，这样我是能收到通知。

**Reply by timongh** [reply]
刚刚又测试了下，只有我直接 fork 你的仓库，并且在我这边才能向你提交 pr。

从你那边进入只会被重定向到官方仓库，而且还搜不到你的用户名。
